laser_encoders/models.py

```diff
@@ -148,6 +153,10 @@ def batch(tokens, lengths, indices):
             yield batch(batch_tokens, batch_lengths, batch_indices)
 
     def encode_sentences(self, sentences):
+        if self.spm_model:
```

Comment:
I don't like two facts fact about this design: 
1) that the same function `encode_sentences` now may expect two types of inputs (either raw or tokenized sentences), and the correct expected type depends on whether `spm_model` was passed to the constructor arguments or not. This may be very confusing to the user.  Can we instead create another method, like `tokenize_and_encode_sentences` or simply `__call__`, that works on raw sentences, and keep the behaviour of `encode_sentences` unchanged>

2) We create a tokenizer each time when encoding the sentences; it would make more sense to initialize it in the encoder's constructor.

Reply:
I get your point 1 butt i don't understand point 2

Reply:
Point 2 is to move the code `tokenizer = LaserTokenizer(spm_model=Path(self.spm_model))` (which is slow) from the `encode_sentences` method (that can be executed multiple times per runtime) to the `__init__` method (that is executed only once).

Reply:
I am done with this